### PR TITLE
make sure domain strings are only sent once

### PR DESCRIPTION
**Pull request by derekwbrown**
### What does this PR do?

consolidates all domain strings into a single list, so that duplicates are not sent

### Motivation

more efficient packaging of submission data

### Checklist

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.

### CODE REVIEW

File: pkg/process/checks/net.go [hunk 1]
```diff
@@ -260,16 +295,21 @@ func batchConnections(
...
+			Connections:     batchConns,
+			GroupId:         groupID,
+			GroupSize:       groupSize,
+			ContainerForPid: ctrIDForPID,
+			//EncodedDNS:        dnsEncoder.Encode(batchDNS),
+			DomainDb: &model.DomainDatabase{
+				EncodedDomains: dnsEncoder.EncodeDomainDatabase(namedb),
+			},
+			EncodedDnsLookups: dnsEncoder.EncodeMapped(batchDNS),
+			IndexedDomains:    batchDomainIndexes,
```
Comment (mbakht) [inline pkg/process/checks/net.go:309]: Do we need this for decoding?

File: pkg/process/checks/net_test.go [hunk 2]
```diff
@@ -239,15 +247,52 @@ func TestNetworkConnectionBatchingWithDomains(t *testing.T) {
...
+			assert.Equal(t, val.DnsStatsByQueryType[int32(network.DNSTypeA)].DnsTimeouts, uint32(1))
 		case 2:
-			assert.Equal(t, []string{"foo.com", "", "baz.com"}, connections.Domains)
+			assert.Equal(t, len(domaindb), 2)
+			assert.Contains(t, domaindb, domains[0])
+			assert.Contains(t, domaindb, domains[2])
+			assert.NotContains(t, domaindb, domains[1])
+
+			indexzero := indexOf(domains[0], domaindb)
+			indexdextwo := indexOf(domains[2], domaindb)
```
Comment (mbakht) [inline pkg/process/checks/net_test.go:270]: s/indexdextwo/indextwo?

File: pkg/process/checks/net_test.go [hunk 3]
```diff
@@ -239,15 +247,52 @@ func TestNetworkConnectionBatchingWithDomains(t *testing.T) {
...
 		case 1:
-			assert.Equal(t, []string{"foo.com", "", ""}, connections.Domains)
+			assert.Equal(t, len(domaindb), 1)
+			assert.Equal(t, domains[0], domaindb[0])
+
+			// check for correctness of the data
+			conn := connections.Connections[0]
+			val, ok := conn.DnsStatsByDomainByQueryType[0]
+			assert.True(t, ok)
+			assert.Equal(t, val.DnsStatsByQueryType[int32(network.DNSTypeA)].DnsTimeouts, uint32(1))
```
Comment (mbakht) [inline pkg/process/checks/net_test.go:262]: Can we have some other `DNSType` in the tests in addition to `DNSTypeA`?

File: pkg/process/checks/net.go (hunk 1 above)
Comment (mbakht) [reply pkg/process/checks/net.go:309]: Can you please explain why? Sorry - I may have missed it.

**Comment by derekwbrown** [comment]
Superceded by #8808 (rebase made a mess of this pr)

**Pull request closed without merging**